ingest/validation/validate_metadata.py

```diff
@@ -1247,13 +1253,107 @@ def review_metadata_names(metadata):
+        if len(ontology_tracker[label].keys()) > 1:
+            multiply_assigned.append(label)
+    multiply_assigned.sort()
+    return multiply_assigned
+
+
+def assess_ontology_ids(ids, property_name, metadata):
+    """
+    Check ordered collection of ontology IDs for increasing numeric values
+    """
+    evidence_of_excel_drag = False
+    evidence_of_excel_drag_threshold = 25
+    binned_ids = defaultdict(list)
+    for id in ids:
+        try:
+            ontology_shortname, term_id = re.split("[_:]", id)
+            binned_ids[ontology_shortname].append(term_id)
+        except (ValueError, TypeError):
+            # invalid ontology id will already be flagged as convention error
+            # no need to also mark as ontology error as part of excel drag detection
+            pass
+    for ontology in binned_ids.keys():
+        id_numerics = []
+        incrementation_count = 0
+        for term in binned_ids[ontology]:
+            term_numeric = re.search('(\d)*$', term)
+            id_numerics.append(int(term_numeric.group()))
+        for x, y in zip(id_numerics, id_numerics[1:]):
+            if y - x == 1:
+                incrementation_count += 1
+            elif y - x != 1:
+                incrementation_count = 0
+            if incrementation_count >= evidence_of_excel_drag_threshold:
+                evidence_of_excel_drag = True
+                break
+    return evidence_of_excel_drag
+
+
+def detect_excel_drag(metadata, convention):
+    """ Check if ontology IDs submitted have characteristic excel drag properties
+        Todo1: "excel drag" detection of array-based ontologyID data
+        is lacking (need to track pipe-delimited string)
+        Todo2: need to bypass EBI OLS queries to "fill in"
+        missing ontology labels for optional metadata)
+        Hint: try working with raw metadata.file data, would
+        allow this check to be moved before collect_jsonschema_errors
+    """
+    excel_drag = False
+    for property_name in metadata.ontology.keys():
+        if len(set(metadata.ordered_ontology[property_name])) == 1:
+            continue
+        else:
+            property_ids = metadata.ordered_ontology[property_name]
+            unique_ids = set(property_ids)
+            property_labels = metadata.ordered_labels[property_name]
+            property_labels_blanks_removed = [i for i in property_labels if i]
+            unique_labels = set(property_labels_blanks_removed)
+            # likely ontology label mis-assignment if multiple ontology IDs ascribed to same ontology label
+            label_multiply_assigned = len(unique_ids) > len(unique_labels)
+            try:
+                if assess_ontology_ids(property_ids, property_name, metadata):
+                    msg = f"{property_name}: incrementing ontology ID values suggest cut and paste issue - exiting validation, ontology content not validated against ontology server.\nPlease confirm ontology IDs are correct and resubmit.\n"
```

Comment:
Non-blocking: I think this error message makes sense to us, but would be confusing to a user.  That being said, I'm not sure what the right message is, so maybe this is ok.

Reply:
Hm, I see what you mean. I'll update to "long stretch(es) of contiguously incrementing" which is a tad more descriptive of the issue.